libs/sdk-core/src/breez_services.rs

```diff
@@ -264,12 +265,15 @@ impl BreezServices {
             });
         }
 
+        let amount_msat = max(provided_amount_msat, invoice_amount_msat);
+
         match self
             .persister
             .get_completed_payment_by_hash(&parsed_invoice.payment_hash)?
         {
             Some(_) => Err(SendPaymentError::AlreadyPaid),
             None => {
+                self.persist_new_bolt11_payment_attempt(&parsed_invoice, amount_msat)?;
```

Comment:
NIT: `persisten_pending_payment` ?

Reply:
Method renamed to `persist_pending_payment` in [d01b703](https://github.com/breez/breez-sdk/pull/621/commits/d01b703f7b0ca40762b57604fa40c35fffd007a2). I assumed the `en` was a typo :) please let me know if it's not.